examples/applications/restapi/txstatewebsocket.py

```diff
@@ -0,0 +1,120 @@
+# Adapted from https://stackoverflow.com/questions/48695294/how-can-i-detect-the-closure-of-an-python-aiohttp-web-socket-on-the-server-when
+# while looking at ways to handle the issue of when the client exits uncleanly
+import json
+import uuid
+import asyncio
+import aiohttp
+from aiohttp import web
+from bitcoinx import hex_str_to_hash
+
+from electrumsv.logs import logs
+from electrumsv.restapi import Fault
+from electrumsv.wallet import AbstractAccount
+
+from examples.applications.restapi.handler_utils import VNAME
+from examples.applications.restapi.errors import Errors
+
+
+class WSClient(object):
+
+    def __init__(self, ws_id: str, websocket: web.WebSocketResponse, account: AbstractAccount):
+        self.ws_id = ws_id
+        self.websocket = websocket
+        self.account = account
+
+
+class TxStateWebSocket(web.View):
+    """
+    Todo: test that Basic Auth covers this endpoint too
+    1) initial registration of a txid begets an initial response of current tx state
+    2) app.py:RESTAPIApplication._tx_state_push_notification gives push notifications for
+    subsequent state changes
+    """
+    logger = logs.get_logger("tx-state-websocket")
+
+    async def get(self):
+        ws = web.WebSocketResponse()
+        await ws.prepare(self.request)
+
+        try:
+            self.restapi = self.request.app['restapi']
+            self.argparser = self.restapi.argparser
+            required_vars = [VNAME.WALLET_NAME, VNAME.ACCOUNT_ID]
+            vars = await self.argparser(self.request, required_vars=required_vars)
+            wallet_name = vars[VNAME.WALLET_NAME]
+            index = vars[VNAME.ACCOUNT_ID]
+
+            await self.restapi._load_wallet(wallet_name)
+            self.account = self.restapi._get_account(wallet_name, index)
+
+            ws_id = str(uuid.uuid4())
+            client = WSClient(ws_id=ws_id, websocket=ws, account=self.account)
+            self.request.app['ws_clients'][client.ws_id] = client
+            self.logger.debug('%s connected. host=%s.' % (client.ws_id, self.request.host))
+            try:
+                await self._handle_new_txid_registration(client)
+            finally:
+                await ws.close()
+                del self.request.app['ws_clients'][client.ws_id]
+                self.logger.debug('%s disconnected' % client.ws_id)
+            return ws
+        except Fault as e:
```

Comment:
Can you reduce the scope of this except clause to the method that raises it?

Reply:
Not really. There are a number of possible `Fault` type exceptions (which all all to do with the restapi's internal helper/utility functions)